libs/blocks/aside/aside.js

```diff
@@ -24,23 +28,59 @@ function decorateLayout(el) {
   return foreground;
 }
 
-function decorateContent(el, isInline) {
+function decorateContent(el, type, size) {
   if (!el) return;
-  const text = el.querySelector('h1, h2, h3, h4, h5, h6')?.closest('div');
+  const text = el.querySelector('h1, h2, h3, h4, h5, h6, a')?.closest('div');
+  const picture = text?.querySelector('picture');
+  const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
   text?.classList.add('text');
+  if (text && !text.querySelector('p')) {
+    const buttons = text.querySelectorAll('em a, strong a');
+    let btnWrap = null;
+    if (buttons[0] && !buttons[0].closest('p')) {
+      btnWrap = document.createElement('p');
+      buttons.forEach((button) => btnWrap.append(button.parentElement));
+    }
+    const desc = createTag('p', null, text.innerHTML);
+    text.innerHTML = '';
+    if (iconArea) text.append(iconArea);
+    text.append(desc);
+    if (btnWrap) text.append(btnWrap);
+  }
+  iconArea?.classList.add('icon-area');
+  decorateButtons(el);
   const headings = text?.querySelectorAll('h1, h2, h3, h4, h5, h6');
   const heading = headings?.[headings.length - 1];
-  heading?.classList.add(`${isInline ? 'heading-S' : 'heading-XL'}`);
-  heading?.nextElementSibling?.classList.add(`${isInline ? 'body-M' : 'body-S'}`);
-  heading?.previousElementSibling?.classList.add('detail-M');
+  const isInline = type === 'inline';
+  const isNotification = type === 'notification';
+  let headingClass = 'heading-XL';
+  let bodyClass = 'body-S';
+  if (isNotification && (size === 'extra-small' || size === 'small')) {
+    headingClass = '';
+  }
+  if ((isNotification && (size === 'extra-small' || size === 'small' || size === 'large')) || isInline) {
+    bodyClass = 'body-M';
+  }
+  if ((isNotification && size === 'medium') || isInline) {
+    headingClass = 'heading-S';
+  }
+  if (isNotification && size === 'large') {
+    headingClass = 'heading-L';
+  }
+  heading?.classList.add(headingClass);
+  const bodyCopy = heading?.nextElementSibling.classList.length === 0 ? heading.nextElementSibling : text?.querySelector('p:not([class])');
+  bodyCopy?.classList.add(bodyClass);
+  const body = createTag('div', { class: 'body-area' });
+  bodyCopy.insertAdjacentElement('beforebegin', body);
+  body.append(bodyCopy);
+  const prevClasses = heading?.previousElementSibling?.classList;
+  if (prevClasses?.length === 0) prevClasses.add('detail-M');
   el.querySelector(':scope > div:not([class])')?.classList.add('image');
```

Comment:
I know it's not changed in the PR, but seems like there could be a better selector than "any div without a class is an image"

Reply:
Updated to `el.querySelector(':scope > div:not(.text) img')?.closest('div').classList.add('image');` It's a little round-about but at least more specific